homeassistant/components/sensor/syncthru.py

```diff
@@ -0,0 +1,244 @@
+    vol.Optional(
+        CONF_MONITORED_CONDITIONS,
+        default=DEFAULT_MONITORED_CONDITIONS
+    ): vol.All(cv.ensure_list, [vol.In(DEFAULT_MONITORED_CONDITIONS)])
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the SyncThru component."""
+    from pysyncthru import SyncThru, test_syncthru
+
+    if discovery_info is not None:
+        host = discovery_info.get(CONF_HOST)
+        name = discovery_info.get(CONF_NAME, DEFAULT_NAME)
+        _LOGGER.debug("Discovered a new Samsung Printer: %s", discovery_info)
+        # Test if the discovered device actually is a syncthru printer
+        if not test_syncthru(host):
+            _LOGGER.error("No SyncThru Printer found at %s", host)
+            return False
+        monitored = DEFAULT_MONITORED_CONDITIONS
+    else:
+        host = config.get(CONF_RESOURCE)
+        name = config.get(CONF_NAME)
+        monitored = config.get(CONF_MONITORED_CONDITIONS)
+
+    # Main device, always added
+    printer = SyncThru(host)
+    printer.update()
+    devices = [SyncThruMain(hass, printer, name)]
+
+    for key in printer.tonerStatus(filter_supported=True):
+        if 'toner_' + str(key) in monitored:
+            devices.append(SyncThruToner(hass, printer, name, key))
+    for key in printer.drumStatus(filter_supported=True):
+        if 'drum_' + str(key) in monitored:
+            devices.append(SyncThruDrum(hass, printer, name, key))
+    for key in printer.inputTrayStatus(filter_supported=True):
+        if 'tray_' + str(key) in monitored:
+            devices.append(SyncThruInputTray(hass, printer, name, key))
+    for key in printer.outputTrayStatus():
+        if 'output_tray_' + str(key) in monitored:
+            devices.append(SyncThruOutputTray(hass, printer, name, key))
+
+    add_devices(devices, True)
+    return True
+
+
+class SyncThruSensor(Entity):
+    """Implementation of an abstract Samsung Printer sensor platform."""
+
+    def __init__(self, hass, syncthru, name):
+        """Initialize the sensor."""
+        self._hass = hass
+        self.syncthru = syncthru
+        self._attributes = {}
+        self._state = STATE_UNKNOWN
+        self._name = name
+        self._friendly_name = name
+        self._icon = 'mdi:printer'
+        self._unit_of_measurement = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Return the icon of the device."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measuremnt."""
+        return self._unit_of_measurement
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        self._attributes[CONF_FRIENDLY_NAME] = self._friendly_name
```

Comment:
Remove this.

Reply:
Friendly name is only for users to set. Platforms can set the `name` property. If platform wants to have different default friendly name from entity_id, the platform sets the entity_id too.